src/Bridge/Doctrine/Orm/Filter/BooleanFilter.php

```diff
@@ -50,18 +54,29 @@ public function apply(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $q
 
         $properties = $this->extractProperties($request);
 
-        foreach ($properties as $property => $boolean) {
-            if (!$this->isPropertyEnabled($property) || !$this->isPropertyMapped($property, $resourceClass)) {
+        foreach ($properties as $property => $value) {
+            if (
+                !$this->isPropertyEnabled($property) ||
+                !$this->isPropertyMapped($property, $resourceClass) ||
+                !$this->isBooleanField($property, $resourceClass)
+            ) {
                 continue;
             }
 
-            if (empty($boolean) && isset($this->properties[$property])) {
-                $boolean = $this->properties[$property];
-            }
-
-            $filterBoolean = filter_var($boolean, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE);
```

Comment:
`filter_var` just hides the complexity. We don't need / want to have to support all those weird cases (including the case insensitivity, and empty string as false).

`( "true" | "false" | "1" | "0" )` is exactly what `xsd:boolean` is, which seems like a good thing to standardize on. This should not be about PHP anyway.